### PR TITLE
build: improve classpath handling

### LINKED ISSUE #1202: build: Pass classpathref to tei:anttasks
As described in https://github.com/TEIC/Stylesheets/issues/544

### PR DESCRIPTION
**Pull request by bwbohl**
* submit classpath to tei ant tasks closes #1202 
* introduce init-mei-classpath target to compile classpath at runtime when needed

### CODE REVIEW

**Comment by musicEnfanthen** [comment]
@bwbohl Suggestion to move this to 5.x, or is it urgently needed to run the build?

**Comment by bwbohl** [comment]
Passing the mei.classpath to the tei ant calls would be really good. This works as it is now but could be improved a bit. I'm happy to delay additional work to 5.x but I would love to see this in 5.0

**Review by rettinghaus — APPROVED** [review]
Let's move forward with this.

**Comment by bwbohl** [comment]
and yes, retested, build will fail without this when run locally, because mei.classpath will be initialized before fetching the libs  (as before) and after this we can just run ant commands without `-lib`property ;-)

File: build.xml [hunk 1]
```diff
@@ -142,6 +136,16 @@
         <antcall target="xerces-download"/>
         <echo>initialized</echo>
     </target>
+    
+    <target name="init-mei-classpath" description="Initialize classpath: initializes the mei.classpath by prepending the jar files contained in the lib directory to the java classpath.">
+        <antcall target="init" if:true="${initialize}"/>
```
Comment (musicEnfanthen) [inline build.xml]: `initialize` variable is never set?

**Comment by musicEnfanthen** [comment]
So the `ant init` step can be skipped at all? And the call with `-lib` ? (BTW, I've run `ant` always without any `-lib` additions...) Could you then please adjust the README accordingly with this PR.

File: build.xml (hunk 1 above)
Comment (bwbohl) [reply build.xml]: good catch will remove this and rely on the docker property

**Comment by bwbohl** [comment]
- [x] change readme
- ~~[ ] change github action~~not necessary
- [x] change ant description

**Comment by bwbohl** [comment]
@musicEnfanthen please double check if your requested changes are now adequately addressed

**Comment by musicEnfanthen** [comment]
Yes, thank you @bwbohl . But now we got some merge conflicts with the lately introduced python changes. Can you resolve?

**Review by rettinghaus — APPROVED** [review]

**Comment by bwbohl** [comment]
conflicts resolved

**Review by rettinghaus — APPROVED** [review]

**Comment by rettinghaus** [comment]
@musicEnfanthen please prioritize. Without this it is not possible to build external customizations.

**Comment by musicEnfanthen** [comment]
Thanks @bwbohl for resolving conflicts.

@rettinghaus Yes, I see . Please note that I am not able to test locally this weekend. 

If needed before monday, it would be great to have someone checkout the PR and test the build properly.

File: BUILD_COMMANDLINE.md [hunk 2]
```diff
@@ -57,76 +57,96 @@
...
+       ant -Dcustomization.path="[/ABSOLUTE/PATH/TO/YOUR/CUSTOMIZATION]" build-compiled-odd
+       ```
+     * Build the RNG schema of a specific customization:
 
-     ```shell
-     ant -lib lib/saxon/saxon-he-11.5.jar -Dcustomization.path="[/ABSOLUTE/PATH/TO/YOUR/CUSTOMIZATION]" build-rng
-     ```
+       ```shell
+       ant -lib lib/saxon/saxon-he-11.5.jar -Dcustomization.path="[/ABSOLUTE/PATH/TO/YOUR/CUSTOMIZATION]" build-rng
+       ```
```
Comment (musicEnfanthen) [inline BUILD_COMMANDLINE.md]: Duplicated step from above? Also, this one here seems to be outdated since lib call is not needed anymore, right?

Comment (bwbohl) [reply BUILD_COMMANDLINE.md]: fixed

**Review by musicEnfanthen — APPROVED** [review]

**Pull request merged**